### DISCUSSION TITLE
Pass criteria for raising and merging a change request to OpenSSL main branch

### DISCUSSION THREAD

**Discussion by quicmmarimut** [openssl/openssl#26063] [Q&A] [answered]
Hi Team,
   We wanted to understand the pass criteria for raising and merging a change request to OpenSSL main branch. I went through the contribution documentation (https://github.com/openssl/openssl/blob/master/CONTRIBUTING.md) mentioned in the wiki section. But did not find any passing or what kind of test cases needs to be executed to be accepted by the community.

Can you please help to share little details about the test cases or test suites that need to be executed for raising and merging a change request to main branch ?

Thanks,
Mallikarjunan

**Comment by Sashan** [comment, marked as answer]
Hello,

if you are adding a new code/feature you should also add tests and plug them to existing tests OpenSSL keeps in `test` subdirectory. Those tests are executed by `make test` command.

if you happen to fix some bug in existing code, the chance is the code is covered by tests already. In this case running `make test` before submitting your PR is the good practice.

lot more testing happens once you submit your PR. the CI process compiles your change on more platforms where also tests run.

did I answer your question?

regards
sashan

**Reply by quicmmarimut** [reply]
Yes, Thank you Sashan !
